src/crypto/tls/handshake_server_tls13.go

```diff
@@ -22,6 +22,45 @@ import (
 // messages cause too much work in session ticket decryption attempts.
 const maxClientPSKIdentities = 5
 
+// EXP_EventServerHandshakeTimingInfo is carries intra-stack durations for
+// state machine changes. It can be used for tracking metrics during a
+// connection.
+//
+// NOTE: This API is EXPERIMENTAL and subject to change.
+type EXP_EventServerHandshakeTimingInfo struct {
+	start                    time.Time
+	ReadClientHello          time.Duration
+	WriteServerHello         time.Duration
+	WriteEncryptedExtensions time.Duration
+	WriteCertificate         time.Duration
+	WriteCertificateVerify   time.Duration
+	WriteServerFinished      time.Duration
+	ReadClientFinished       time.Duration
+}
+
+// Name is required by the EXP_Event interface.
+func (e EXP_EventServerHandshakeTimingInfo) Name() string {
+	return "ServerHandshakeTimingInfo"
+}
+
+func (e EXP_EventServerHandshakeTimingInfo) elapsedTime() time.Duration {
+	return time.Now().Sub(e.start)
+}
+
+const (
```

Comment:
These constants and `infoNames` below can be dropped, as they're not used anywhere else.

Reply:
Yep, missed that!